lib/net.js

```diff
@@ -745,38 +752,16 @@ Socket.prototype._writeGeneric = function(writev, data, encoding, cb) {
     return false;
   }
 
-  var req = new WriteWrap();
-  req.handle = this._handle;
-  req.oncomplete = afterWrite;
-  req.async = false;
-  var err;
-
-  if (writev) {
-    var allBuffers = data.allBuffers;
-    var chunks;
-    var i;
-    if (allBuffers) {
-      chunks = data;
-      for (i = 0; i < data.length; i++)
-        data[i] = data[i].chunk;
-    } else {
-      chunks = new Array(data.length << 1);
-      for (i = 0; i < data.length; i++) {
-        var entry = data[i];
-        chunks[i * 2] = entry.chunk;
-        chunks[i * 2 + 1] = entry.encoding;
-      }
-    }
-    err = this._handle.writev(req, chunks, allBuffers);
+  let ret;
+  const req = new WriteWrap();
+  const handle = this._handle;
+  this[kCreateWriteWrap](req, handle, afterWrite);
+  if (writev)
+    ret = this[kWritevGeneric](req, data, cb);
+  else
+    ret = this[kWriteGeneric](req, data, encoding, cb);
 
-    // Retain chunks
-    if (err === 0) req._chunks = chunks;
-  } else {
-    err = createWriteReq(req, this._handle, data, encoding);
-  }
-
-  if (err)
-    return this.destroy(errnoException(err, 'write', req.error), cb);
+  if (ret) return ret;
```

Comment:
Given the note above re: changed semantics for http2, I would rather we just test for `err` and then call `this.destroy` here rather than inside the helpers. The `if (ret) return ret;` is super opaque as to what the return value is or its meaning.

Reply:
Since @addaleax requested this related change, I rescind the feedback above. That said, a comment should be left indicating that `if (ret) return ret;` means there was an error and `this.destroy` was called with it.

Reply:
@apapirovski Just for context, I talked with @aks- about this and I don’t think it’s a good permanent solution anyway, but doing more might require some changes to the `bytesWritten` tracking for net

But yes, a comment here would be very helpful